dev/tools/dartdoc_checker.dart

```diff
@@ -82,7 +80,7 @@ void checkForUnresolvedDirectives(String htmlOutputPath) {
 int _scanFile(File file) {
   assert(path.extension(file.path) == '.html');
   final Iterable<String> matches = _pattern.allMatches(file.readAsStringSync())
-      .map((RegExpMatch m ) => m.group(0));
+      .map((RegExpMatch m ) => m.group(0)!);
```

Comment:
are we sure `m.group(0)` cannot be `null`?

Reply:
I believe, `group(0)` should always be non-null since it is the `entire match of the pattern` [1] and allMatches above only returns matches.

[1] https://master-api.flutter.dev/flutter/dart-core/Match/group.html

Reply:
As @goderbauer mentioned, using `group(0)` while iterating through all matches makes no harm since it won't iterate if there are no matches.. and if there are then it will return an entire match (which must exist) making it safe to use null assertion operator.